### PR TITLE
Adding Preshower detector (PSR) for ALICE3 simulation

### PR DESCRIPTION
**Pull request by an15ms157**
This PR creates a simple 8 layered preshower detector named PSR. This is a cylindrical detector of length 100 cm whose shower layers are made up of Pb (0.5 cm) and the detector layers are made up of Si (45 microns). PSR is based on the ITSMFT classes and the code is structurally similar to that of FT3. Each layer is made of a monolithic silicon disk with a thin sensitive layer for hit generation. Silicon chip thickness is tuned to match the layer x/X0 to allow a minimal evaluation of material budget effects.

One should get a file o2sim_HitsPSR.root by running
$ o2-sim -m PSR -e TGeant3 -g boxgen -n 10

### CODE REVIEW

**Comment by mconcas** [comment]
ok, now that codechecker is fine the next step is to reabase your branch on  the AliceO2/dev one. Since the `pragma cheker` is something that has been added in relatively recent times wrt your PR opening.

**Comment by an15ms157** [comment]
I have made the changes.

File: Detectors/Base/src/MaterialManager.cxx [hunk 1]
```diff
@@ -292,6 +292,7 @@ void MaterialManager::insertMaterialName(const char* uniquename, int index)
 // inserts the last
 void MaterialManager::insertTGeoMedium(std::string modname, int localindex)
 {
+  LOG(INFO) << "Changes for build";
```
Comment (mconcas) [inline Detectors/Base/src/MaterialManager.cxx]: Perhaps a DEBUG print you forgot?

File: Detectors/Upgrades/ALICE3/FT3/simulation/src/Detector.cxx [hunk 2]
```diff
@@ -160,7 +160,7 @@ void Detector::buildBasicFT3(const FT3BaseParam& param)
   mLayerID.clear();
   mLayers.resize(2);
 
-  for (Int_t direction : {0, 1}) {
```
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/FT3/simulation/src/Detector.cxx:163]: This is not really clear to me, why these changes in FT3 whould affact the PSR? @rpezzi is it fine to you?

File: Detectors/Upgrades/ALICE3/PSR/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<!-- doxy
+\page refDetectorsUpgradesALICE3FT3 EndCaps
+/doxy -->
+
+# PostLS4EndCaps
```
Comment (mconcas) [inline Detectors/Upgrades/ALICE3/PSR/README.md:5]: This is minor, but I think should be changed to desccribe PSR

**Review by mconcas — CHANGES_REQUESTED** [review]
To me looks pretty good for the part of the upgrades, some inline comments.

File: Detectors/Base/src/MaterialManager.cxx (hunk 1 above)
Comment (an15ms157) [reply Detectors/Base/src/MaterialManager.cxx]: Ah!! yes

**Comment by mconcas** [comment]
@an15ms157 : do you have any update on this? Could you also rebase on `dev` branch, so that the git history is linear? Thanks

**Comment by mconcas** [comment]
Hi @an15ms157, you should rebase the modifications you want to merge on the dev branch. The history must be linear, e.g. no merging commits and branching. Could you please do that?

**Comment by an15ms157** [comment]
Hello @mconcas 
I am not sure that I have as of yet managed to make the required changes. So, I have tried to make a new local branch, clump all the content to a single commit and push it to my fork. Here is it: https://github.com/an15ms157/AliceO2/tree/mydev
Is it possible that I run another PR with this different branch? In case if such is possible and it is sucessful, I will keep that one and remove this PR.

**Comment by mconcas** [comment]
> Hello @mconcas
> I am not sure that I have as of yet managed to make the required changes. So, I have tried to make a new local branch, clump all the content to a single commit and push it to my fork. Here is it: https://github.com/an15ms157/AliceO2/tree/mydev
> Is it possible that I run another PR with this different branch? In case if such is possible and it is sucessful, I will keep that one and remove this PR.

Hi @an15ms157: I had a look to your new branch and it seems to be ok, please proceed with a new PR. Thanks.

**Comment by an15ms157** [comment]
> > Hello @mconcas
> > I am not sure that I have as of yet managed to make the required changes. So, I have tried to make a new local branch, clump all the content to a single commit and push it to my fork. Here is it: https://github.com/an15ms157/AliceO2/tree/mydev
> > Is it possible that I run another PR with this different branch? In case if such is possible and it is sucessful, I will keep that one and remove this PR.
> 
> Hi @an15ms157: I had a look to your new branch and it seems to be ok, please proceed with a new PR. Thanks.
Hello @mconcas .Here is a link to my new PR: https://github.com/AliceO2Group/AliceO2/pull/6727

**Pull request closed without merging**